lib/utils/search_debouncer.dart

```diff
@@ -0,0 +1,24 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+
+class Debouncer {
+  final Duration duration;
```

Comment:
All the attributes could be private?

Reply:
duration can be private, the other two are accessed outside the class so cannot be private.

Reply:
Have made all of them private, used a getter to get the value of one of the fields.